#### Libraries

In [20]:
#Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from tqdm import tqdm

#Options
pd.options.display.float_format = '{:.2f}'.format

***

#### Datasets

In [21]:
#Datasets

df1 = pd.read_csv(r"C:\Users\gcuci\Desktop\Proyecto\EDA Steam\data\steamspy_2019.csv")
df2 = pd.read_csv(r"C:\Users\gcuci\Desktop\Proyecto\EDA Steam\data\steamspy_2020.csv")
df3 = pd.read_csv(r"C:\Users\gcuci\Desktop\Proyecto\EDA Steam\data\steamspy_2021.csv")

#Main df
df = pd.concat(map(pd.read_csv, [r"C:\Users\gcuci\Desktop\Proyecto\EDA Steam\data\steamspy_2019.csv",\
    r"C:\Users\gcuci\Desktop\Proyecto\EDA Steam\data\steamspy_2020.csv",\
        r"C:\Users\gcuci\Desktop\Proyecto\EDA Steam\data\steamspy_2021.csv"]), ignore_index=True)

In [22]:
df.head()

Unnamed: 0            id                           name  \
0           0   /app/632470  Disco Elysium - The Final Cut   
1           1   /app/620980                     Beat Saber   
2           2  /app/1174180          Red Dead Redemption 2   
3           3   /app/601150                Devil May Cry 5   
4           4   /app/883710                Resident Evil 2   

                            dev             publisher  \
0                     ['ZA/UM']             ['ZA/UM']   
1                ['Beat Games']        ['Beat Games']   
2            ['Rockstar Games']    ['Rockstar Games']   
3  ['CAPCOM Co., Ltd.', 'Site']  ['CAPCOM Co., Ltd.']   
4          ['CAPCOM Co., Ltd.']  ['CAPCOM Co., Ltd.']   

                          genre  \
0                       ['RPG']   
1  ['Indie', 'in Early Access']   
2       ['Action', 'Adventure']   
3                    ['Action']   
4                    ['Action']   

                                            language  \
0  ['English', 'Simplified Chinese', 'Spanish - S...   
1  ['English', 'French', 'German', 'Spanish - Spa...   
2  ['English', 'French', 'Italian', 'German', 'Sp...   
3  ['English', 'French', 'Italian', 'German', 'Sp...   
4  ['English', 'French', 'Italian', 'German', 'Sp...   

                                                tags  \
0  ['RPG', 'Story Rich', 'Choices Matter', 'Detec...   
1  ['VR', 'Rhythm', 'Music', 'Great Soundtrack', ...   
2  ['Open World', 'Adventure', 'Story Rich', 'Wes...   
3  ['Action', 'Hack and Slash', 'Great Soundtrack...   
4  ['Zombies', 'Survival Horror', 'Horror', 'Rema...   

                                            category           release  \
0  ['Single-player', 'Steam Achievements', 'Full ...  ['Oct 15, 2019']   
1  ['Single-player', 'Multi-player', 'PvP', 'Onli...  ['May 21, 2019']   
2  ['Single-player', 'Multi-player', 'PvP', 'Onli...   ['Dec 5, 2019']   
3  ['Single-player', 'Multi-player', 'Co-op', 'On...   ['Mar 7, 2019']   
4  ['Single-player', 'Steam Achievements', 'Full ...  ['Jan 24, 2019']   

  actual_price discount old_userscore metascore                      owners  \
0   ['$13.99']  ['65%']       ['94%']   ['97%']   ['1,000,000 - 2,000,000']   
1   ['$29.99']       []       ['96%']   ['93%']   ['2,000,000 - 5,000,000']   
2   ['$29.99']  ['50%']       ['88%']   ['93%']  ['5,000,000 - 10,000,000']   
3    ['$9.89']  ['67%']       ['94%']   ['89%']   ['2,000,000 - 5,000,000']   
4   ['$15.99']  ['60%']       ['97%']   ['89%']  ['5,000,000 - 10,000,000']   

     followers  
0  ['185,290']  
1  ['149,465']  
2  ['525,455']  
3  ['245,562']  
4  ['282,997']

In [23]:
df.dtypes

Unnamed: 0        int64
id               object
name             object
dev              object
publisher        object
genre            object
language         object
tags             object
category         object
release          object
actual_price     object
discount         object
old_userscore    object
metascore        object
owners           object
followers        object
dtype: object

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27877 entries, 0 to 27876
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     27877 non-null  int64 
 1   id             27877 non-null  object
 2   name           27875 non-null  object
 3   dev            27876 non-null  object
 4   publisher      27876 non-null  object
 5   genre          27876 non-null  object
 6   language       27876 non-null  object
 7   tags           27876 non-null  object
 8   category       27876 non-null  object
 9   release        27876 non-null  object
 10  actual_price   27876 non-null  object
 11  discount       27876 non-null  object
 12  old_userscore  27876 non-null  object
 13  metascore      27876 non-null  object
 14  owners         27876 non-null  object
 15  followers      27876 non-null  object
dtypes: int64(1), object(15)
memory usage: 3.4+ MB


***

#### Finding nulls

In [25]:
#Finding nulls
df[df["name"].isnull()]

Unnamed: 0            id name        dev publisher  \
5111        5111  /app/1116910  NaN       ['']      ['']   
8085         287  /app/1195390  NaN  ['', ' ']     [' ']   

                                                  genre language  \
5111  ['Action', 'Adventure', 'Casual', 'Indie', 'RP...       []   
8085                                               ['']       []   

                                                   tags  \
5111  ['Casual', 'Adventure', 'Indie', 'Strategy', '...   
8085                                                 []   

                              category           release actual_price  \
5111  ['Single-player', 'Steam Cloud']  ['Sep 25, 2019']    ['$6.99']   
8085                              ['']  ['Mar 14, 2020']           []   

     discount old_userscore metascore          owners followers  
5111       []            []        []  ['0 - 20,000']   ['565']  
8085       []            []        []  ['0 - 20,000']   ['185']

In [26]:
#Finding nulls
df[df["dev"].isnull()]

Unnamed: 0            id                                name  dev  \
24277        6937  /app/1718040  Clarence Goes to the F&#%ING Store  NaN   

      publisher genre language tags category release actual_price discount  \
24277       NaN   NaN      NaN  NaN      NaN     NaN          NaN      NaN   

      old_userscore metascore owners followers  
24277           NaN       NaN    NaN       NaN

***

#### Fixing df


In [27]:
#Fixing df
df.dropna(inplace = True)
df.drop(["Unnamed: 0"], axis = 1, inplace = True)
changes_in_rows = ["\[", "\]", "\'", "%", "\$", "/app/"]
for i in changes_in_rows:
    df.replace(i, "", regex = True, inplace = True)
df.reset_index(drop = True, inplace = True)

***

#### Changing types in columns

In [28]:
#Changing types in columns
df["release"] = pd.to_datetime(df["release"])
df["discount"] = df["discount"].replace("",0).astype(int)
df["actual_price"] = df["actual_price"].replace("",0).astype(float)
df["old_userscore"] = df["old_userscore"].replace("",0).astype(int)
df["metascore"] = df["metascore"].replace("",0).astype(int)
df["followers"] = df["followers"].str.replace(",","").replace("",0).astype(int)

In [29]:
df.dtypes

id                       object
name                     object
dev                      object
publisher                object
genre                    object
language                 object
tags                     object
category                 object
release          datetime64[ns]
actual_price            float64
discount                  int32
old_userscore             int32
metascore                 int32
owners                   object
followers                 int32
dtype: object

***

#### New columns

In [30]:
#New columns

#Initial price
df["initial_price"] = df["actual_price"] / (1 - ( df["discount"] / 100))
df["initial_price"] = df["initial_price"].round()
df.replace([np.inf, -np.inf], 0, inplace = True)
column_to_move = df.pop("initial_price")
df.insert(9, "initial_price", column_to_move)
df['initial_price'] = df['initial_price'].astype(int)

#Genre_lst, colunma de genero guardado como una lista
df["genre_lst"] = df["genre"].replace(', ', ',', regex= True, inplace= True)
df["genre_lst"] = df.genre.apply(lambda x: x[:].split(","))

#Sales to category
conditions = [(df["owners"] == "50,000,000 - 100,000,000"), (df["owners"] == "20,000,000 - 50,000,000"),\
    (df["owners"] == "10,000,000 - 20,000,000"), (df["owners"] == "5,000,000 - 10,000,000"),\
        (df["owners"] == "2,000,000 - 5,000,000"), (df["owners"] == "1,000,000 - 2,000,000"),\
            (df["owners"] == "500,000 - 1,000,000"), (df["owners"] == "200,000 - 500,000"),\
                (df["owners"] == "100,000 - 200,000"), (df["owners"] == "50,000 - 100,000"),\
                    (df["owners"] == "20,000 - 50,000"), (df["owners"] == "0 - 20,000")]

category = [1,2,3,4,5,6,7,8,9,10,11,12]

mean_sales = [75000000, 35000000, 15000000, 7500000, 3500000, 1500000, 750000, 350000, 150000, 75000, 35000, 10000]

#Mean sales 
df["mean_sales"] = np.select(conditions, mean_sales)
column_to_move = df.pop("mean_sales")
df.insert(11, "mean_sales", column_to_move)

#Total sales in usd
df["total_sales"]= df["mean_sales"] * df["initial_price"]
column_to_move = df.pop("total_sales")
df.insert(12, "total_sales", column_to_move)

#Category sales
df["category_sales"] = np.select(conditions, category)
column_to_move = df.pop("category_sales")
df.insert(13, "category_sales", column_to_move)

***

####  Failed attempt

In [31]:
#Altering Amazon sales New world\
# (Intente cambiar el valor de la venta del juego New World de Amazon pero no consegui la forma, dice que el valor es muy grande)
df.drop([183], axis = 0 , inplace = True)
df.reset_index(drop = True, inplace = True)

***

#### Variables

In [32]:
#Variables

#Genders_lst
genre_lst = pd.Series(np.concatenate(df.genre_lst))
genre_total = genre_lst.value_counts().sort_index().rename_axis("genre")\
    .reset_index(name = "count")

#DF whiout free to play games
free_to_play = df["initial_price"] != 0
df_whitouth_free = df[free_to_play]

#Total of games
total_games = df.shape[0]

df_owners_sales = df[["owners", "category_sales", "mean_sales"]].sort_values\
    (by = "category_sales").drop_duplicates()

#DF with Indies
df_indie = df[df["genre"].str.contains("Indie")]

#DF without Indies
df_not_indie = df[~df["genre"].str.contains("Indie")]

#Juegos que superaron las 100K copias
more_than_100k = df[df["category_sales"] <= 9].shape[0]

#Juegos que vendieron mas de un millon de copias
more_than_1_million = df[df["category_sales"] <= 6].shape[0]

#Total juegos Indie
total_indies = df_indie.shape[0]

#Total juegos AAA
total_AAA = df_not_indie.shape[0]

#Ventas de videojuegos por dia
sales_per_day = df["mean_sales"].sum()

***

#### Eliminated games from Steam

In [33]:
#games without publisher, eliminated from steam(88)
df[df["publisher"] == ""].head()

id                                name                      dev  \
242   499660               Medieval Kingdom Wars  , Reverie World Studios   
291   811500             The Last Piece of Bacon         , LisaDanceGames   
310  1143470      Vengeance ~ In my familys name                            
313  1148630                           Hot Takes                            
317  1160740  Mermaid Coralie ~ Love and Madness                            

    publisher                                             genre  \
242            Action,Indie,Simulation,Strategy,in Early Access   
291                  Action,Adventure,Casual,Indie,Early Access   
310                                                Casual,Indie   
313                                                Casual,Indie   
317                                                Casual,Indie   

                                              language  \
242  English, German, Portuguese - Brazil, French, ...   
291                                            English   
310                                    English, French   
313                                    English, French   
317                                    English, French   

                                                  tags  \
242  Grand Strategy, RTS, City Builder, 4X, Base-Bu...   
291                                                      
310                                                      
313                                                      
317                                                      

                                              category    release  \
242  Single-player, Multi-player, PvP, Online PvP, ... 2019-01-03   
291                                      Single-player 2019-08-03   
310  Single-player, Steam Achievements, Full contro... 2019-08-30   
313  Single-player, Steam Achievements, Partial Con... 2019-09-09   
317  Single-player, Steam Achievements, Full contro... 2019-10-02   

     initial_price  actual_price  mean_sales  total_sales  category_sales  \
242             20          7.99      150000      3000000               9   
291              0          0.00       10000            0              12   
310              0          0.00       10000            0              12   
313              0          0.00       10000            0              12   
317              0          0.00       10000            0              12   

     discount  old_userscore  metascore             owners  followers  \
242        60             77         60  100,000 - 200,000      53979   
291         0              0          0         0 - 20,000         10   
310         0              0          0         0 - 20,000         24   
313         0              0          0         0 - 20,000         26   
317         0              0          0         0 - 20,000         24   

                                             genre_lst  
242  [Action, Indie, Simulation, Strategy, in Early...  
291   [Action, Adventure, Casual, Indie, Early Access]  
310                                    [Casual, Indie]  
313                                    [Casual, Indie]  
317                                    [Casual, Indie]

***

#### Final df

In [34]:
#Result
df.head()

id                           name                     dev  \
0   632470  Disco Elysium - The Final Cut                   ZA/UM   
1   620980                     Beat Saber              Beat Games   
2  1174180          Red Dead Redemption 2          Rockstar Games   
3   601150                Devil May Cry 5  CAPCOM Co., Ltd., Site   
4   883710                Resident Evil 2        CAPCOM Co., Ltd.   

          publisher                  genre  \
0             ZA/UM                    RPG   
1        Beat Games  Indie,in Early Access   
2    Rockstar Games       Action,Adventure   
3  CAPCOM Co., Ltd.                 Action   
4  CAPCOM Co., Ltd.                 Action   

                                            language  \
0  English, Simplified Chinese, Spanish - Spain, ...   
1  English, French, German, Spanish - Spain, Japa...   
2  English, French, Italian, German, Spanish - Sp...   
3  English, French, Italian, German, Spanish - Sp...   
4  English, French, Italian, German, Spanish - Sp...   

                                                tags  \
0  RPG, Story Rich, Choices Matter, Detective, Is...   
1  VR, Rhythm, Music, Great Soundtrack, Moddable,...   
2  Open World, Adventure, Story Rich, Western, Ac...   
3  Action, Hack and Slash, Great Soundtrack, Char...   
4  Zombies, Survival Horror, Horror, Remake, Sing...   

                                            category    release  \
0  Single-player, Steam Achievements, Full contro... 2019-10-15   
1  Single-player, Multi-player, PvP, Online PvP, ... 2019-05-21   
2  Single-player, Multi-player, PvP, Online PvP, ... 2019-12-05   
3  Single-player, Multi-player, Co-op, Online Co-... 2019-03-07   
4  Single-player, Steam Achievements, Full contro... 2019-01-24   

   initial_price  actual_price  mean_sales  total_sales  category_sales  \
0             40         13.99     1500000     60000000               6   
1             30         29.99     3500000    105000000               5   
2             60         29.99     7500000    450000000               4   
3             30          9.89     3500000    105000000               5   
4             40         15.99     7500000    300000000               4   

   discount  old_userscore  metascore                  owners  followers  \
0        65             94         97   1,000,000 - 2,000,000     185290   
1         0             96         93   2,000,000 - 5,000,000     149465   
2        50             88         93  5,000,000 - 10,000,000     525455   
3        67             94         89   2,000,000 - 5,000,000     245562   
4        60             97         89  5,000,000 - 10,000,000     282997   

                  genre_lst  
0                     [RPG]  
1  [Indie, in Early Access]  
2       [Action, Adventure]  
3                  [Action]  
4                  [Action]

In [35]:
#TEMPORAL
df.to_csv("steam_df_tableau")

***

# EDA: Mercado de videojuegos en Steam

***

#### 1) Cuáles fueron los juegos más vendidos y sus copias?	

In [36]:
df_whitouth_free[["name", "owners", "category_sales", "total_sales"]].sort_values(by = "category_sales", ascending = True)\
    .reset_index(drop = True).head(20)

name                    owners  \
0                                  New World  50,000,000 - 100,000,000   
1                                    Valheim   20,000,000 - 50,000,000   
2                      鬼谷八荒 Tale of Immortal   10,000,000 - 20,000,000   
3                               Phasmophobia   10,000,000 - 20,000,000   
4                             Cyberpunk 2077   10,000,000 - 20,000,000   
5               Mount & Blade II: Bannerlord   10,000,000 - 20,000,000   
6                         NARAKA: BLADEPOINT   10,000,000 - 20,000,000   
7                           Battlefield 2042    5,000,000 - 10,000,000   
8                             Sea of Thieves    5,000,000 - 10,000,000   
9                                   Factorio    5,000,000 - 10,000,000   
10                            Risk of Rain 2    5,000,000 - 10,000,000   
11         Halo: The Master Chief Collection    5,000,000 - 10,000,000   
12  Sekiro: Shadows Die Twice - GOTY Edition    5,000,000 - 10,000,000   
13                           Resident Evil 2    5,000,000 - 10,000,000   
14                     Red Dead Redemption 2    5,000,000 - 10,000,000   
15                                     Hades    5,000,000 - 10,000,000   
16                         Vampire Survivors     2,000,000 - 5,000,000   
17                           Forza Horizon 4     2,000,000 - 5,000,000   
18                              Metro Exodus     2,000,000 - 5,000,000   
19                        Deep Rock Galactic     2,000,000 - 5,000,000   

    category_sales  total_sales  
0                1  -1294967296  
1                2    700000000  
2                3    300000000  
3                3    210000000  
4                3    900000000  
5                3    750000000  
6                3    300000000  
7                4    450000000  
8                4    300000000  
9                4    225000000  
10               4    187500000  
11               4    300000000  
12               4    450000000  
13               4    300000000  
14               4    450000000  
15               4    187500000  
16               5     10500000  
17               5    210000000  
18               5    105000000  
19               5    105000000

% de los juegos más vendidos que son Indie

In [37]:
round((8 / 20) *100, 2)

40.0

Como podemos apreciar los juegos más vendidos en su mayoría son AAA con 60% de las ventas, pero no podemos dejar atrás a los juegos Indie ya que poseen un 40% de las ventas del mercado de Steam, lo que sugiera que el mercado no tiene una fuerte preferencia por uno u otro.

***

#### 2) Datos de los juegos más vendidos?

In [38]:
df_whitouth_free["total_sales"] = df["mean_sales"] * df["initial_price"]

C:\Users\gcuci\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [39]:
df_whitouth_free.sort_values(by = "category_sales", ascending = True).reset_index(drop = True)\
    .head(20).describe()[["total_sales"]]

total_sales
count          20.00
mean    257276635.20
std     429468012.86
min   -1294967296.00
25%     187500000.00
50%     300000000.00
75%     450000000.00
max     900000000.00

La venta promedio de los juegos más vendidos (20), es de 329 millones de dólares, mientras que sus percentiles son:
25%: 184 millones,
50%: 300 millones,
75%: 450 millones,
max: 900 millones. Lo que demuestra lo lucrativo que puede llegar a ser este mercado.

***

#### 3) Cuáles fueron los géneros más populares?

In [40]:
df_genre = genre_total.sort_values(by = "count", ascending = False).reset_index(drop = True)
df_genre[(df_genre["genre"] != ("Early Access")) & (df_genre["genre"] != ("in Early Access")) & (df_genre["genre"]!= ("Free to Play"))]\
    .head(10).reset_index(drop = True)

genre  count
0                  Indie  20526
1                 Casual  12176
2                 Action  11540
3              Adventure  10944
4             Simulation   5651
5               Strategy   5297
6                    RPG   5062
7                 Sports   1278
8                 Racing   1005
9  Massively Multiplayer    709

Los géneros más populares son bastante destacables, en su mayoría son Indies, seguidos de Casual, Action, Adventure, Simulation.
Cabe destacar con los videojuegos se categorizan en su mayoría en más de un género por ello, la suma del total de los géneros es mayor que el total de juegos.

***

#### 4) Cuáles fueron los géneros menos populares?

In [41]:
genre_total.sort_values(by = "count", ascending = False).reset_index(drop = True).tail(10)

genre  count
21  Software Training     68
22      Photo Editing     48
23     Web Publishing     47
24         Accounting      9
25              Short      1
26              Movie      1
27           Tutorial      1
28              360 V      1
29           Episodic      1
30        Documentary      1

No es ninguna sorpresa que entre los géneros menos atractivos en la tienda son los que no tienen nada que ver con los juegos en general, como software de edición.

***

#### 5) Cuántos juegos superaron las 100k copias?

Total de juegos publicados

In [42]:
total_games

27873

Total de todos juegos

In [43]:
more_than_100k

1636

% de juegos

In [44]:
round((more_than_100k / total_games) * 100, 2)

5.87

Esto evidencia que el número de juegos totales y su porcentaje en es mínimo en comparación al total de juegos publicados, lo que quiere decir que es un mercado bien competitivo. Solo 1635 superaron las 100K copias, lo que representa una muestra de menos del 6% del total.

***

Total juegos Indie

In [45]:
total_indies

20526

Total de juegos Indie que superaron los 100K

In [46]:
df_indie[df_indie["category_sales"] <= 9].shape[0]

1046

% de juegos los juegos Indie que superaron los 100k 

In [47]:
round((1046 / total_indies) * 100, 2)

5.1

En el caso de los juegos Indie el porcentaje es muy similar a la media de todos los juegos en cuanto a superar la brecha de las 100K copias, menos del 6%.

***

Total juegos AAA


In [48]:
total_AAA

7347

Total de juegos AAA que superaron los 100K

In [49]:
df_not_indie[df_not_indie["category_sales"] <= 9].shape[0]

590

% de los juegos AAA que superaron los 100K

In [50]:
round((589 / total_AAA) * 100, 2)

8.02

En el caso de los AAA podemos observar una mejora en los porcentajes, pero la diferencia no es abismal, solo un 8% de ellos superaron los 100K.

***

#### 6) Cuántos juegos superaron el millón de copias?

Todos los juegos

In [51]:
more_than_1_million

187

% de todos los juegos que superaron el millon de copias

In [52]:
round((more_than_1_million / total_games) * 100, 2)

0.67

En este caso solo 186 juegos, superaron el millón de copias, siendo el porcentaje menor al 1% (0,67%) del total de juegos, lo que indica que solo unos pocos llegan a ser extremadamente rentables.

***

Total de juegos Indie que superaron el millon de copias

In [53]:
df_indie[df_indie["category_sales"] <= 6].shape[0]

75

% de juegos Indie que superaron el millon de copias

In [54]:
round((75 / total_indies) * 100, 2) 

0.37

En el caso de los juegos Indie, solo 75 llegaron a superar el millón de copias, lo que da un porcentaje similar, solo el 0,37% de todos los juegos Indies llegan a superar dicha cifra.

***

Total de juegos AAA que superaron el millon de copias

In [55]:
df_not_indie[df_not_indie["category_sales"] <= 6].shape[0]

112

% de los juegos AAA que superaron el millon de copias

In [56]:
round((111 / total_AAA) * 100, 2)

1.51

En el caso de los juegos AAA el total fue de 111 juegos, lo que representa el 1,51% de todos los AAA, lo que sugiera una mejora con respecto a los Indies.

***

#### 7) Cuáles fueron los juegos Indie más vendidos?

In [57]:
df_indie[["name", "owners", "total_sales"]].sort_values(by = "total_sales", ascending = False).head(20).reset_index(drop = True)

name                   owners  total_sales
0                              Valheim  20,000,000 - 50,000,000    700000000
1                鬼谷八荒 Tale of Immortal  10,000,000 - 20,000,000    300000000
2                             Factorio   5,000,000 - 10,000,000    225000000
3                         Phasmophobia  10,000,000 - 20,000,000    210000000
4                                Hades   5,000,000 - 10,000,000    187500000
5                       Risk of Rain 2   5,000,000 - 10,000,000    187500000
6                                Squad    2,000,000 - 5,000,000    175000000
7              Wolcen: Lords of Mayhem    2,000,000 - 5,000,000    140000000
8                         Satisfactory    2,000,000 - 5,000,000    105000000
9                              MORDHAU    2,000,000 - 5,000,000    105000000
10                           ASTRONEER    2,000,000 - 5,000,000    105000000
11                          Beat Saber    2,000,000 - 5,000,000    105000000
12                      Slay the Spire    2,000,000 - 5,000,000     87500000
13                 Oxygen Not Included    2,000,000 - 5,000,000     87500000
14  Pathfinder: Wrath of the Righteous    1,000,000 - 2,000,000     75000000
15                     Space Engineers    2,000,000 - 5,000,000     70000000
16                          Black Mesa    2,000,000 - 5,000,000     70000000
17                Dyson Sphere Program    2,000,000 - 5,000,000     70000000
18                      Gunfire Reborn    2,000,000 - 5,000,000     70000000
19   Totally Accurate Battle Simulator    2,000,000 - 5,000,000     70000000

Sin duda alguna cuando se llegan a desarrollar juegos realmente interesantes y bien llevados a cabo, se puede conseguir grandes retornos, destacando que en una producción Indie es mucho más rentables que una AAA debido a sus costes más bajos. Dicho rango de ventas va desde los 2 millones de copias, hasta los 50 millones.

***

#### 8) Cuál es la facturación promedio de los juegos Indie más vendidos?

In [58]:
df_indie[["total_sales"]].sort_values(by = "total_sales", ascending = False).head(20).reset_index(drop = True).describe()

total_sales
count        20.00
mean  157250000.00
std   143315762.90
min    70000000.00
25%    73750000.00
50%   105000000.00
75%   187500000.00
max   700000000.00

Los juegos Indies más vendidos (20), en promedio tienen una media de ventas de 157 millones de dólares, mientras que sus percentiles son:
25%: 73 millones,
50%: 105 millones,
75%: 187 millones,
max: 700 millones. Lo que demuestra lo lucrativo que puede llegar a ser este mercado.


***

#### 9) Cuál es la facturación promedio de los juegos AAA más vendidos?

In [59]:
df_not_indie[["total_sales"]].sort_values(by = "total_sales", ascending = False).head(20).reset_index(drop = True).describe()

total_sales
count        20.00
mean  323750000.00
std   194035231.47
min   175000000.00
25%   210000000.00
50%   210000000.00
75%   337500000.00
max   900000000.00

Los AAA destacan al momento de facturación total, donde en promedio facturan el doble que los indies, con un promedio de 323 millones, mientras que sus percentiles son:
25%: 210 millones,
50%: 210 millones,
75%: 337 millones,
max: 900 millones. Lo que demuestra lo lucrativo que puede llegar a ser este mercado.

***

#### 10) Cuáles fueron los Publisher que más juegos han sacado a la venta?

In [60]:
df["publisher"].value_counts().head(10)

8floor                150
Big Fish Games        120
HH-Games              113
                       88
Laush Studio           67
Sokpop Collective      65
Piece Of Voxel         64
Kagura Games           63
Boogygames Studios     54
William at Oxford      51
Name: publisher, dtype: int64

Como podemos apreciar, en su mayoría son Publisher pequeños, como es el caso de William at Oxford, aunque cabe resaltar grandes casas como 8floor.
(Nota: el espacio vacío son juegos que han sido eliminados de Steam).

***

#### 11) Cuáles fueron los Publisher con las mayores ventas?

In [61]:
df.groupby(["publisher"])[["total_sales"]].sum().sort_values(by = "total_sales", ascending= False).head(10)

total_sales
publisher                                         
Electronic Arts                         1815275000
Xbox Game Studios                       1525000000
Coffee Stain Publishing                  911500000
CD PROJEKT RED                           900000000
CAPCOM Co., Ltd.                         779550000
TaleWorlds Entertainment                 750000000
2K                                       665000000
Rockstar Games                           550000000
SEGA                                     490600000
Activision (Excluding Japan and Asia)    450000000

Aquí vemos un caso totalmente diferente, donde los todos los Publisher con más ventas son todos AAA, no se encuentra ninguna casa pequeña, y donde las ventas mínimas van desde los 450 millones, hasta los 1815 millones de dólares de facturación que genero Electronic Arts.

***

#### 12) Cuáles fueron los developers que más juegos han sacado a la venta?

In [62]:
df["dev"].value_counts().head(10)

Creobit, 8floor                           71
                                          67
Sokpop Collective                         64
Boogygames Studios                        54
William at Oxford                         51
Laush Dmitriy Sergeevich, Laush Studio    46
Choice of Games                           44
玫瑰工作室                                     40
Blender Games                             40
Somer Games, 8floor                       39
Name: dev, dtype: int64

Aquí se puede apreciar algo muy curioso, hay developers que han llegado a crear hasta 71 juegos en tres años, lo que lo vuelve un poco extraño, dado a que la mayoría son developers pequeños y desconocidos.

***

#### 13) Cuáles fueron los developers con las mayores ventas?

In [63]:
df.groupby(["dev", "owners"])[["total_sales"]].sum().sort_values(by = "total_sales", ascending= False).head(10)

total_sales
dev                                                owners                              
CD PROJEKT RED                                     10,000,000 - 20,000,000    900000000
TaleWorlds Entertainment                           10,000,000 - 20,000,000    750000000
Iron Gate AB, Coffee Stain Publishing              20,000,000 - 50,000,000    700000000
Activision (Excluding Japan and Asia), FromSoft... 5,000,000 - 10,000,000     450000000
Electronic Arts, DICE                              5,000,000 - 10,000,000     450000000
Rockstar Games                                     5,000,000 - 10,000,000     450000000
Xbox Game Studios, Playground Games                2,000,000 - 5,000,000      420000000
DICE, Electronic Arts                              2,000,000 - 5,000,000      315000000
Ruffian Games, Splash Damage, Xbox Game Studios... 5,000,000 - 10,000,000     300000000
24 Entertainment, NetEase Games Montréal           10,000,000 - 20,000,000    300000000

Como no es sorpresa, se vuelve a ver que los developers con mayores ganancias son todos AAA, y resalta mucho la atención en el caso de CD Projekt Red donde es publicador y distribuidor a la vez, lo que le permite quedarse con toda la facturación, que son unos 900 millones de dólares. El que menos vendió de los que más vende en la lista llego a los 300 millones.

***

#### 14) Cuántos juegos se publican al día?

In [64]:
total_games / (365 * 3)

25.454794520547946

Es alarmante el número de juegos que se publican al día en Steam, lo que vuelve difícil el poder sobresalir en este mercado / tienda de videojuegos.

***

#### 15) Cuántos juegos se han vendido en promedio al día en Steam?

In [65]:
sales_per_day / (365 * 3)

1505219.1780821919

En promedio al día se venden 1.4 millones de juegos en Steam.

In [66]:
sales_per_day

1648215000

En total en los últimos 3 años en Steam se han vendido 1.5 billones de juegos, muy impresionante.

In [67]:
df_indie["mean_sales"].sum()

850835000

Los Indies representan el 54% de las ventas totales.

In [68]:
df_not_indie["mean_sales"].sum()

797380000

Los triple AAA representan el 46% de las ventas.

***

#### 16) Cuál es el mes con más reléase?

In [69]:
df_date = pd.to_datetime(df["release"]).dt.month

In [70]:
df_date["month"] = df["release"].dt.month_name()

c:\Users\gcuci\AppData\Local\Programs\Python\Python37\lib\site-packages\pandas\core\series.py:1056: SettingWithCopyWarning: modifications to a property of a datetimelike object are not supported and are discarded. Change values on the original.
  cacher_needs_updating = self._check_is_chained_assignment_possible()
c:\Users\gcuci\AppData\Local\Programs\Python\Python37\lib\site-packages\pandas\core\indexing.py:1724: SettingWithCopyWarning: modifications to a property of a datetimelike object are not supported and are discarded. Change values on the original.
  self._setitem_with_indexer_missing(indexer, value)


Todos los juegos

In [71]:
df_date["month"].value_counts()

October      2767
December     2516
November     2465
August       2426
September    2399
May          2297
July         2279
March        2221
April        2213
February     2201
June         2089
January      2000
Name: release, dtype: int64

Los meses con mayores reléase se producen el segundo semestre, para ser más específico, octubre, noviembre y diciembre. Este es el caso de todos los videojuegos.

***

Indies

In [72]:
df_indie["month"] = df["release"].dt.month_name()

C:\Users\gcuci\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [73]:
df_indie["month"].value_counts()

October      2042
August       1815
November     1801
December     1779
September    1778
May          1728
April        1675
July         1659
February     1648
March        1602
January      1507
June         1492
Name: month, dtype: int64

En el caso de los Indie, ocurren exactamente lo mismo, solo que el mes con los mayores reléase es octubre.

***

AAA

In [74]:
df_not_indie["month"] = df["release"].dt.month_name()

C:\Users\gcuci\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [79]:
df_not_indie["month"].value_counts()

December     737
October      725
November     664
September    621
July         620
March        619
August       611
June         597
May          569
February     553
April        538
January      493
Name: month, dtype: int64

En el caso de los AAA los mayores reléase se producen en la misma fecha que todos los videojuegos en promedio, siendo diciembre el mes con mayores publicaciones.

***

#### 17) Cuál es el descuento más utilizado por los Publisher?

In [80]:
df["discount"].describe()

count   27873.00
mean       27.82
std        30.06
min         0.00
25%         0.00
50%        20.00
75%        50.00
max       100.00
Name: discount, dtype: float64

El descuento en promedio es del 27%, y el máximo se ubica al 100%.

In [81]:
df["discount"].value_counts().sort_values(ascending = False).head(10)

0     13071
50     3006
75     1219
51     1075
30     1055
40     1020
60      982
90      809
20      696
80      692
Name: discount, dtype: int64

Y el descuento que más se repite es del 50%, seguido del 75% y 71% respectivamente; Cabe resaltar que casi la mitad de los videojuegos (13.071) no posee algún descuento, interesante, considerando que estamos en promociones de verano.

***

#### 18) Shovelgames? El caso William at Oxford

Curiosidad del EDA:

Cabe resaltar que esto era totalmente desconocido para mí, existe un concepto llamado Shovelgames, lo que quiere decir que son juegos que se realizan en dos o tres días, es decir de baja calidad, que son publicados de una manera masiva para buscar generar algunas ventas, un caso muy claro y que llamo la atención es William at Oxford. Es tan grande el problema que Steam en los últimos años ha eliminado miles de juegos para "tratar" de solucionar este problema.


***